src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -153,15 +169,24 @@ const PlrDaoStakingTransactionBlock = ({
   const [totalKeyBasedPLRTokens, setTotalKeyBasedPLRTokens] = useState<number>(0);
   const [totalSmartWalletPLRTokens, setTotalSmartWalletPLRTokens] = useState<number>(0);
 
+  const [selectedRoute, setSelectedRoute] = useState<SelectOption | null>(null);
+  const [availableRoutes, setAvailableRoutes] = useState<Route[] | null>(null);
+  const [isLoadingAvailableRoutes, setIsLoadingAvailableRoutes] = useState<boolean>(false);
+
   const [accounts, setAccounts] = useState<AccountBalance[]>([]);
   const [isNFTMember, setIsNFTMember] = useState<boolean>(false);
 
   const hasEnoughPLR =
     totalKeyBasedPLRTokens >= MAX_PLR_TOKEN_LIMIT || totalSmartWalletPLRTokens >= MAX_PLR_TOKEN_LIMIT;
-
-  const filteredSupportedChains = supportedChains.filter((chain) =>
-    [CHAIN_ID.ETHEREUM_MAINNET, CHAIN_ID.POLYGON, CHAIN_ID.XDAI, CHAIN_ID.BINANCE].includes(chain.chainId)
+  const enoughPolygonPLR = accounts.some((account) =>
+    account.chainName === 'Polygon' && selectedAccountType === 'Contract'
+      ? account.smartWallet >= MAX_PLR_TOKEN_LIMIT
+      : account.keyBasedWallet >= MAX_PLR_TOKEN_LIMIT
   );
+  const isPolygonAccountWithEnoughPLR =
```

Comment:
how is this different from `enoughPolygonPLR` (which has comments already too)? imo you should combine this into single